File: EPSILON/TOOLS/Altova/Altova/XMLSpy2006/spl/java/altova/text/edifact/EDIFactSettings.java
[call GenerateFileHeader("EDIFactSettings.java")]
package com.altova.text.edifact;

public class EDIFactSettings extends EDISettings {
	private long mSyntaxVersionNumber = 2;

	private char mSyntaxLevel = 'A';

	private String mControllingAgency = "UNO";

	private boolean mWriteUNA = true;

	public long getSyntaxVersionNumber() {
		return mSyntaxVersionNumber;
	}

	public char getSyntaxLevel() {
		return mSyntaxLevel;
	}

	public String getControllingAgency() {
		return mControllingAgency;
	}

	public boolean getWriteUNA() {
		return mWriteUNA;
	}

	public void setSyntaxVersionNumber(long rhs) {
		mSyntaxVersionNumber = rhs;
	}

	public void setSyntaxLevel(char rhs) {
		mSyntaxLevel = rhs;
	}

	public void setControllingAgency(String rhs) {
		mControllingAgency = rhs;
	}

	public void setWriteUNA(boolean rhs) {
		mWriteUNA = rhs;
	}
}

File: EPSILON/TOOLS/Altova/Altova/XMLSpy2006/spl/cpp/AltovaText/EDIFactSettings.cpp
[call GenerateFileHeader("EDIFactSettings.cpp")]
#include "StdAfx.h"
#include "EDIFactSettings.h"

namespace altova
{
namespace text
{
namespace edi
{

CEDIFactSettings::CEDIFactSettings()
:	m_SyntaxVersionNumber(2)
,	m_SyntaxLevel(_T('A'))
,	m_ControllingAgency(_T("UNO"))
,	m_WriteUNA(true)
{}
long CEDIFactSettings::GetSyntaxVersionNumber() const
{
	return m_SyntaxVersionNumber;
}
TCHAR CEDIFactSettings::GetSyntaxLevel() const
{
	return m_SyntaxLevel;
}
const tstring& CEDIFactSettings::GetControllingAgency() const
{
	return m_ControllingAgency;
}
bool CEDIFactSettings::GetWriteUNA() const
{
	return m_WriteUNA;
}
void CEDIFactSettings::SetSyntaxVersionNumber(long rhs)
{
	m_SyntaxVersionNumber= rhs;
}
void CEDIFactSettings::SetSyntaxLevel(TCHAR rhs)
{
	m_SyntaxLevel= rhs;
}
void CEDIFactSettings::SetControllingAgency(const tstring& rhs)
{
	m_ControllingAgency= rhs;
}
void CEDIFactSettings::SetWriteUNA(bool rhs)
{
	m_WriteUNA= rhs;
}

} // namespace edi
} // namespace text
} // namespace altova
